packages/devtools_app/lib/src/inspector/layout_explorer/flex/flex.dart

```diff
@@ -452,6 +364,149 @@ class _FlexLayoutExplorerWidgetState extends LayoutExplorerWidgetState<
+          final selectedRenderObject =
+              selectedChildKey.currentContext?.findRenderObject();
+          if (selectedRenderObject != null &&
+              widget.scrollController.hasClients) {
+            widget.scrollController.position.ensureVisible(
+              selectedRenderObject,
+              alignment: 0.5,
+              duration: defaultDuration,
+            );
+          }
+        });
+      }
+    }
+    if (!widget.properties.hasChildren) {
+      return const Center(child: Text('No Children'));
+    }
+
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+
+    final contents = Container(
+      decoration: BoxDecoration(
+        border: Border.all(
+          color: theme.primaryColorLight,
+        ),
+      ),
+      margin: const EdgeInsets.only(top: margin, left: margin),
+      child: LayoutBuilder(builder: (context, constraints) {
+        final maxWidth = constraints.maxWidth;
+        final maxHeight = constraints.maxHeight;
+
+        double maxSizeAvailable(Axis axis) {
+          return axis == Axis.horizontal ? maxWidth : maxHeight;
+        }
+
+        final childrenAndMainAxisSpacesRenderProps =
+            widget.properties.childrenRenderProperties(
+          smallestRenderWidth: minRenderWidth,
+          largestRenderWidth: defaultMaxRenderWidth,
+          smallestRenderHeight: minRenderHeight,
+          largestRenderHeight: defaultMaxRenderHeight,
+          maxSizeAvailable: maxSizeAvailable,
+        );
+
+        final renderProperties = childrenAndMainAxisSpacesRenderProps
+            .where((renderProps) => !renderProps.isFreeSpace)
+            .toList();
+        final mainAxisSpaces = childrenAndMainAxisSpacesRenderProps
+            .where((renderProps) => renderProps.isFreeSpace)
+            .toList();
+        final crossAxisSpaces = widget.properties.crossAxisSpaces(
+          childrenRenderProperties: renderProperties,
+          maxSizeAvailable: maxSizeAvailable,
+        );
+
+        final childrenRenderWidgets = <Widget>[];
+        for (var i = 0; i < widget.children.length; i++) {
+          final child = widget.children[i];
+          final isSelected = widget.highlighted == child;
+
+          childrenRenderWidgets.add(FlexChildVisualizer(
+            key: isSelected ? selectedChildKey : null,
+            state: widget.state,
+            layoutProperties: child,
+            isSelected: isSelected,
+            renderProperties: renderProperties[i],
+          ));
+        }
+
+        final freeSpacesWidgets = [
+          for (var renderProperties in [...mainAxisSpaces, ...crossAxisSpaces])
+            FreeSpaceVisualizerWidget(renderProperties),
+        ];
+        return Scrollbar(
+          isAlwaysShown: true,
+          controller: widget.scrollController,
+          child: SingleChildScrollView(
+            scrollDirection: widget.properties.direction,
+            controller: widget.scrollController,
```

Comment:
except for this line to specify the scrollController so that ScrollView doesn't get confused and scrollbars appear on the first frame and are interactive.